File: app/src/main/java/com/my/televip/virtuals/tgnet/NativeByteBuffer.java
package com.my.televip.virtuals.tgnet;

import com.my.televip.Class.ClassLoad;
import com.my.televip.Class.ClassNames;
import com.my.televip.obfuscate.ArgsResolver;
import com.my.televip.obfuscate.Obfuscate;

import de.robv.android.xposed.XposedHelpers;

public class NativeByteBuffer {

    public final Object nativeByteBuffer;

    public NativeByteBuffer(Object obj){
        nativeByteBuffer = obj;
    }
    public NativeByteBuffer(boolean calculate){
        nativeByteBuffer = XposedHelpers.newInstance(ClassLoad.getClass(ClassNames.NATIVE_BYTE_BUFFER), calculate);
    }

    public void reuse(){
        XposedHelpers.callMethod(nativeByteBuffer, Obfuscate.getMethodName("NativeByteBuffer", "reuse"));
    }

    public int readInt32(boolean exception) {
        return (int) XposedHelpers.callMethod(nativeByteBuffer, Obfuscate.getMethodName("NativeByteBuffer","readInt32"), exception);
    }

    public void position(int i) {
        XposedHelpers.callMethod(nativeByteBuffer, Obfuscate.getMethodName("NativeByteBuffer","position"), i);
    }

    public void writeInt32(int i) {
        XposedHelpers.callMethod(nativeByteBuffer, Obfuscate.getMethodName("NativeByteBuffer","writeInt32"), i);
    }


}

File: app/src/main/java/com/my/televip/virtuals/tgnet/NativeByteBuffer.kt
package com.my.televip.virtuals.tgnet

import com.my.televip.Class.ClassLoad
import com.my.televip.Class.ClassNames
import com.my.televip.obfuscate.Obfuscate
import de.robv.android.xposed.XposedHelpers

class NativeByteBuffer {

    @JvmField
    var nativeByteBuffer: Any? = null

    constructor(obj: Any?) {
        nativeByteBuffer = obj
    }

    constructor(calculate: Boolean) {
        val nbClass = ClassLoad.getClass(ClassNames.NATIVE_BYTE_BUFFER)
        nativeByteBuffer = XposedHelpers.newInstance(nbClass, calculate)
    }

    fun reuse() {
        val target = nativeByteBuffer ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("NativeByteBuffer", "reuse"))
    }

    fun readInt32(exception: Boolean): Int {
        val target = nativeByteBuffer ?: return 0
        return XposedHelpers.callMethod(target, Obfuscate.getMethodName("NativeByteBuffer", "readInt32"), exception) as? Int ?: 0
    }

    fun position(i: Int) {
        val target = nativeByteBuffer ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("NativeByteBuffer", "position"), i)
    }

    fun writeInt32(i: Int) {
        val target = nativeByteBuffer ?: return
        XposedHelpers.callMethod(target, Obfuscate.getMethodName("NativeByteBuffer", "writeInt32"), i)
    }
}
